Index: qooxdoo/tool/pylib/generator/resource/ImageClipping.py
===================================================================
--- qooxdoo/tool/pylib/generator/resource/ImageClipping.py	(revision 26020)
+++ qooxdoo/tool/pylib/generator/resource/ImageClipping.py	(working copy)
@@ -153,7 +153,8 @@
         temp = os.fdopen(fileDescriptor, "w")
         temp.write("\n".join(clips))
         temp.close()
-        cmd = montage_cmd % (orientation, "@" + os.path.basename(tempPath), combined)
+        #cmd = montage_cmd % (orientation, "@" + os.path.basename(tempPath), combined)
+        cmd = montage_cmd % (orientation, " ".join(clips), combined)
         rc = os.system(cmd)
         os.unlink(tempPath)
         if rc != 0:
